### PR TITLE
Transformer NMT

### PR DESCRIPTION
**Pull request by hirofumi0810**
- Add Transformer-baed NMT model
- Visualize token strings (not index) for attention plotting

### CODE REVIEW

**Comment by hirofumi0810** [comment]
New plotting function generate a figure like this:
ted_00767_0032400_0041270 decoder decoders 0 src_attn ep 200

**Comment by sw005320** [comment]
> New plotting function generates a figure like this:
> ted_00767_0032400_0041270 decoder decoders 0 src_attn ep 200

it looks very good.
asr output or tts input may follow the same way.
btw, is it possible to make the input part better looking by avoiding overlaps?

**Comment by hirofumi0810** [comment]
@sw005320 It is possible to rotate the source tokens, but it takes some time to check it.

File: egs/must_c/mt1/conf/tuning/train_pytorch_transformer.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+lsm-weight: 0.1
+
+# minibatch related
+batch-size: 64  # for BPE
+maxlen-in: 100  # if input length  > maxlen-in, batchsize is automatically reduced
+maxlen-out: 100 # if output length > maxlen-out, batchsize is automatically reduced
+
+# optimization related
+sortagrad: 0 # Feed samples from shortest to longest ; -1: enabled for all epochs, 0: disabled, other: enabled for 'other' epochs
+opt: noam
```
Comment (butsugiri) [inline egs/must_c/mt1/conf/tuning/train_pytorch_transformer.yaml]: Question: what are noam and transformer-lr for? Is vanilla Adam not good?
In Transformer-NMT, I think that we usually use Adam with warmup https://github.com/pytorch/fairseq/tree/master/examples/scaling_nmt

Comment (hirofumi0810) [reply egs/must_c/mt1/conf/tuning/train_pytorch_transformer.yaml]: Noam in ESPnet means Adam with lr scheduling based on the original Transformer paper.

Comment (hirofumi0810) [reply egs/must_c/mt1/conf/tuning/train_pytorch_transformer.yaml]: And also the hyperparameter such as beta is modified.

Comment (butsugiri) [reply egs/must_c/mt1/conf/tuning/train_pytorch_transformer.yaml]: I see. Do you think adding a description for that is a good idea?

Comment (sw005320) [reply egs/must_c/mt1/conf/tuning/train_pytorch_transformer.yaml]: good idea! maybe, we should add this in the main readme. I'll do it with another PR.

**Comment by sw005320** [comment]
> @sw005320 It is possible to rotate the source tokens, but it takes some time to check it.

ok, you don't have to do it in this PR, but it would be great if you do it in the near future.

**Comment by sw005320** [comment]
thanks!
